### ISSUE TITLE
v6 changed the migrations behaviour

### ISSUE THREAD

**Issue by roderik** [ConsenSys-archive/ganache-cli-archive#414]
in v5 if I ran "truffle migrate" on a new testrpc run, it was able to figure out that the Migrations contract was no longer there and do a full deploy.

since v6 it fails like this:

```
Error: Invalid JSON RPC response: {"id":3,"jsonrpc":"2.0"}
    at Object.InvalidResponse (/home/travis/.nvm/versions/node/v8.9.0/lib/node_modules/truffle/build/cli.bundled.js:43303:16)
    at /home/travis/.nvm/versions/node/v8.9.0/lib/node_modules/truffle/build/cli.bundled.js:331156:36
    at /home/travis/.nvm/versions/node/v8.9.0/lib/node_modules/truffle/build/cli.bundled.js:314196:9
    at XMLHttpRequest.request.onreadystatechange (/home/travis/.nvm/versions/node/v8.9.0/lib/node_modules/truffle/build/cli.bundled.js:315621:13)
    at XMLHttpRequestEventTarget.dispatchEvent (/home/travis/.nvm/versions/node/v8.9.0/lib/node_modules/truffle/build/cli.bundled.js:70159:18)
    at XMLHttpRequest._setReadyState (/home/travis/.nvm/versions/node/v8.9.0/lib/node_modules/truffle/build/cli.bundled.js:70449:12)
    at XMLHttpRequest._onHttpResponseEnd (/home/travis/.nvm/versions/node/v8.9.0/lib/node_modules/truffle/build/cli.bundled.js:70604:12)
    at IncomingMessage.<anonymous> (/home/travis/.nvm/versions/node/v8.9.0/lib/node_modules/truffle/build/cli.bundled.js:70564:24)
    at emitNone (events.js:111:20)
    at IncomingMessage.emit (events.js:208:7)
```

Running with --reset fixes this, but it is a change vs the previous version.

I'll cross open an issue at Truffle to see where the fix should happen.

**Comment by benjamincburns** [comment]
@roderik thanks for reporting! I'll look into it shortly!

**Comment by benjamincburns** [comment]
@roderik I wasn't able to reproduce this using the following steps - can you please let me know if this still reproduces for you on v6.0.1 with Truffle 4?

1. `mkdir test_project`
2. `cd test_project && truffle unbox metacoin`
3. Edited `truffle.js` to have the below contents
4. In another console ran `testrpc` (v6.0.1) with no arguments
5. Ran `truffle migrate` in the first console.
6. Restarted `testrpc`
7. Ran `truffle migrate` again in the first console

Result: no error was produced.

My `truffle.js` file:

```javascript
module.exports = {
  networks: {
    development: {
      gas: 6000000,
      host: "localhost",
      port: 8545,
      network_id: "*" // Match any network id
    }
  }
};
```

**Comment by benjamincburns** [comment]
@roderik just kidding - we managed to reproduce it.

**Comment by roderik** [comment]
😏  I had to go back to v5 due to the crashing issue #417

**Comment by benjamincburns** [comment]
@roderik Thanks heaps for reporting this one - this should be fixed by testrpc v6.0.2, which was published just now.
